core/src/main/codegen/templates/Parser.jj

```diff
@@ -6297,6 +6297,18 @@ SqlNode BuiltinFunctionCall() :
                             RESOURCE.illegalFromEmpty());
                     }
                 }
+                e = Expression(ExprContext.ACCEPT_SUB_QUERY) {
+                    if (flag == null) {
+                        flag = SqlTrimFunction.Flag.BOTH.symbol(SqlParserPos.ZERO);
+                    }
+                }
+                <RPAREN> {
+                    args = startList(flag);
+                    args.add(trimChars);
+                    args.add(e);
+                    return SqlStdOperatorTable.TRIM.createCall(s.end(this), args);
+                }
+
             |
                 <RPAREN> {
                     // This is to handle the case of TRIM(x)
```

Comment:
In this path, you need to add a check that throws an error if trimChars is uninitialized, I pushed a test that demonstrates the issue I'm talking about.

Reply:
Actually, on second glance, this is an issue that seemed to already exist. But we should still fix it here.